File: frontend/src/types/sculpture.ts
export interface SculptureAnalysis {
  id: string;
  projectId?: string;
  asset_id?: string;
  imageUrl?: string;
  fileName?: string;
  iconography: string[];
  motifs: string[];
  materials?: string[];
  style_traits?: string[];
  composition_notes?: string;
  period?: string;
  summary?: string;
  confidence: number;
  createdAt?: string;
  created_at?: string;
  metadata?: SculptureMetadata;
}

export interface SculptureMetadata {
  dimensions?: string;
  estimatedPeriod?: string;
  style?: string;
  region?: string;
  condition?: string;
}

File: backend/app/schemas/sculpture.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class SculptureAnalysisResponse(BaseModel):
    id: str
    asset_id: str
    iconography: List[str]
    motifs: List[str]
    materials: Optional[List[str]] = []
    period: Optional[str] = None
    summary: Optional[str] = None
    composition_notes: Optional[str]
    style_traits: List[str]
    confidence: float
    similar_sculptures: Optional[List[dict]] = []
    created_at: datetime
    
    class Config:
        from_attributes = True

